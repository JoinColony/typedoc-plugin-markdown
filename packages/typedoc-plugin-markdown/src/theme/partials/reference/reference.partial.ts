import { ReferenceReflection } from 'typedoc';
import { MarkdownThemeRenderContext } from '../../theme.context';

export function referencePartial(
  context: MarkdownThemeRenderContext,
  props: ReferenceReflection,
) {
  const referenced = props.tryGetTargetReflectionDeep();

  if (!referenced) {
    return `Re-exports ${props.name}`;
  }

  if (props.name === referenced.name) {
    return `Re-exports [${referenced.name}](${context.relativeURL(
      referenced.url,
    )})`;
  }

  return `Renames and re-exports [${referenced.name}](${context.relativeURL(
    referenced.url,
  )})`;
}
