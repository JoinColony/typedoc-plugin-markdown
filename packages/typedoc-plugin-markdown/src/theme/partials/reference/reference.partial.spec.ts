import { ProjectReflection, ReferenceReflection } from 'typedoc';
import { referencePartial } from './reference.partial';

describe(`Reference Partial:`, () => {
  let project: ProjectReflection;

  beforeAll(async () => {
    project = global.getProject('members.ts');
  });

  test(`should compile a referenced member'`, () => {
    expect(
      referencePartial(
        global.getMockContext(),
        project.findReflectionByName('ReferencedMember') as ReferenceReflection,
      ),
    ).toMatchSnapshot();
  });
});
