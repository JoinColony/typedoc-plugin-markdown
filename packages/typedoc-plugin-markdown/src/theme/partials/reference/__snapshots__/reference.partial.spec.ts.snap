// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reference Partial: should compile a referenced member' 1`] = `"Renames and re-exports [ReferencedClass](( relativeURL ))"`;
